### PR TITLE
[debugger] Debugger tries to execute a single step when the thread is already terminated

### PR DESCRIPTION
**Pull request by thaystg**
I couldn't create a small test case that reproduces, I reproduced debugging (Xamarin.Mac) ios-sim-sharp.

Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/942093

### CODE REVIEW

File: mono/mini/debugger-agent.c [hunk 1]
```diff
@@ -7266,6 +7266,14 @@ event_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 				return err;
 			}
 
+			DebuggerTlsData *tls = (DebuggerTlsData *)mono_g_hash_table_lookup (thread_to_tls, THREAD_TO_INTERNAL(step_thread));
+			
+			if (tls->terminated) {
```
Comment (lambdageek) [inline mono/mini/debugger-agent.c]: Can `tls` be null here?

Comment (thaystg) [reply mono/mini/debugger-agent.c]: Yes, you are right, I changed how to get and the inserted the assert like is done in other places.

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Pull request merged**